### ISSUE TITLE
Refactor payments-server API calls to use axios-hooks rather than Redux?

### ISSUE THREAD

**Issue by lmorchard** [mozilla/fxa#3526]
In payments-server, we use Redux with promise & thunk middleware to handle HTTP API calls & call sequences. Redux actions & state track them through pending, success, and failure states. This results in a lot of paperwork for each API call - actions to trigger & track calls, state properties to manage call status, etc.

I started reading about [axios-hooks](https://github.com/simoneb/axios-hooks) recently - and it looks like this library could handle just about everything we're doing for API calls with Redux but with significantly less paperwork.

We could probably reduce the whole API-handling system down to one module exporting prefab per-call hooks. Maybe use AppContext to share around globally-used data like plans, subscriptions, etc. We might even be able to excise Redux from the project altogether, since we haven't ended up using it for anything but this API-tracking function.

**Comment by lmorchard** [comment]
More info from a Medium post: https://medium.com/better-programming/the-modern-way-to-use-promise-based-http-requests-axios-hooks-f00791345a37

**Comment by LZoog** [comment]
With how we're using Redux in payments, I do think this could really simplify things. I've used axios (not axios-hooks) and found it easy to use, and I once interviewed someone that raved about replacing their store completely with `Context` so this is a good thing to keep on our radar. I suppose we just need to consider if/when we could need Redux in the future.

**Comment by lmorchard** [comment]
Yeah, we could also replace the API handling with axios and keep Redux around for what it's good for. The current system of API calls through Redux seemed (to me) like a good idea at the time, but now it honestly looks a bit burdensome since seeing something like axios-hooks.

Since hooks arrived in general, we could replace a lot of Redux functionality with useReducer + useContext. Though, we'd lose things like easy middleware and Redux dev tools.

**Comment by lmorchard** [comment]
Poking at this a bit today and I realized one thing that axios-hooks is missing: There's no way to reset the fetch state. So, if we POST to an API and get an error result, there's no way (as far as I can tell) to reset the request on error dialog dismiss to try again.

But, this might still be a good idea, even if we have to write our own local version of a fetch hook.

**Comment by data-sync-user** [comment]
➤ Jared Hirsch commented:

[~lorchard@mozilla.com]Is this worth keeping around?

**Comment by lmorchard** [comment]
Probably redundant with #5683, but it would be very nice if payments-server could drop Redux someday
